zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -145,6 +145,10 @@
     private ZooKeeperServerShutdownHandler zkShutdownHandler;
     private volatile int createSessionTrackerServerId = 1;
 
+    private static volatile long flushDelay = Long.getLong("zookeeper.flushDelay", 0);
```

Comment:
Can we have these variables as instance variables?

Reply:
Them being static is in support of the values being alterable via jmx. Without this, the admin is stuck resetting the values every time a new ZooKeeperServer is instantiated. 

In my mind, it's a question of how to support this style of setting: via static variables close to the code that uses them or via non-static mutable config (such as QuorumPeerConfig).